File: ChineseAuctionAngular/ChineseAuctionAngular/src/app/models/OrderDTO.ts
// 1. ה-Enum חייב להופיע ראשון ומחוץ ל-Class
export enum OrderStatus {
  Draft = 0,
  Completed = 1
}

// 2. הגדרת מתנה בתוך הזמנה
export class OrdersGiftDTO {
  category: string = '';
  name: string = '';
  amount: number = 1;
  price: number = 0;
  description: string = '';
  image?: string;

  constructor(init?: Partial<OrdersGiftDTO>) {
    Object.assign(this, init);
  }
}

// 3. הגדרת חבילה בתוך הזמנה (השדה שגרם לשגיאה בקומפוננטה)
export class OrdersPackageDTO {
  idPackage: number = 0;
  quantity: number = 0;
  name: string = '';
  price: number = 0;

  constructor(init?: Partial<OrdersPackageDTO>) {
    Object.assign(this, init);
  }
}

// 4. אובייקט ההזמנה המלא
export class OrderDTO {
  totalAmount: number = 0;
  totalPrice: number = 0;
  idUser: number = 0;
  amount: number = 1;
  orderDate: Date = new Date();
  status: OrderStatus = OrderStatus.Draft;
  
  // הוספת המערכים
  ordersPackages: OrdersPackageDTO[] = []; 
  ordersGifts: OrdersGiftDTO[] = []; 

  constructor(init?: Partial<OrderDTO>) {
    Object.assign(this, init);
    if (init?.orderDate) {
      this.orderDate = new Date(init.orderDate);
    }
    // אתחול בטוח של המערכים
    this.ordersPackages = init?.ordersPackages ? init.ordersPackages.map(p => new OrdersPackageDTO(p)) : [];
    this.ordersGifts = init?.ordersGifts ? init.ordersGifts.map(g => new OrdersGiftDTO(g)) : [];
  }
}

File: ChineseAuctionAPI/ChineseAuctionAPI/DTOs/OrderDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using ChineseAuctionAPI.Models;

namespace ChineseAuctionAPI.DTOs
{
  

    // 2. הגדרת ה-DTO עבור המתנה בתוך ההזמנה
    public class OrdersGiftDTO
    {
        [Required]
        public string Category { get; set; } = string.Empty;
        [Required]
        public string Name { get; set; } = string.Empty;
        [Range(1, 1000)]
        public int Amount { get; set; } = 1;
        [Required, Range(1, int.MaxValue)]
        public int Price { get; set; }
        [StringLength(500)]
        public string Description { get; set; } = string.Empty;
        public string? Image { get; set; }
    }

    // 3. אובייקט ההזמנה המלא שמחבר את הכל
    public class OrderDTO
    {
        [Required, Range(0, int.MaxValue)]
        public int TotalAmount { get; set; }
        [Required, Range(0, int.MaxValue)]
        public int TotalPrice { get; set; }
        [Required]
        public int IdUser { get; set; }
        [Range(1, 1000)]
        public int Amount { get; set; } = 1;
        public DateTime OrderDate { get; set; } = DateTime.Now;
        public OrderStatus Status { get; set; } = OrderStatus.Draft;

        // הרשימות שמאפשרות לנתונים לעבור לאנגולר
        public List<OrdersPackageDTO> OrdersPackages { get; set; } = new();
        public List<OrdersGiftDTO> OrdersGifts { get; set; } = new();
    }
}